LandSandBoat/server
# Core rework to add Trust iLvl scaling to emulate retail-like conditio…

…ns for HP/MP as well as Stat calculation modifiers



**_I affirm:_**

## What does this pull request do?

- This PR is designed to add an iLvl scaling emulation to Trusts considering Ilvl of the master for mLvl/sLvl adjustments.
- this PR scales HP/MP and modifies Stat levels to aid in trust efficiency from level 99-119. 
- this was taken from retail observations via youtube of the trust Valaineral.
- Valaineral sample data.
![val 99](https://user-images.githubusercontent.com/99501957/196979739-e69835cc-0e61-4590-91b1-74d3c9a88e82.PNG)
![Val 101](https://user-images.githubusercontent.com/99501957/196979749-e4b507ce-1689-4579-8fc4-cf9882ff6072.PNG)
![val 103](https://user-images.githubusercontent.com/99501957/196979752-f33eef10-5389-4703-975f-9dab3d379844.PNG)
![val 105](https://user-images.githubusercontent.com/99501957/196979756-96943605-4502-4216-b283-d2b5406171c9.PNG)
![val 107](https://user-images.githubusercontent.com/99501957/196979758-56f83c36-b45a-42c0-803d-4cf9f0865e91.PNG)
![val 109](https://user-images.githubusercontent.com/99501957/196979895-41e78bc9-a48a-40a7-b4a1-1034de4ef702.PNG)
![val 114](https://user-images.githubusercontent.com/99501957/196979906-8a5572d3-d989-4add-8f2c-5a67726aebeb.PNG)
![val 119](https://user-images.githubusercontent.com/99501957/196979910-7d120f11-4dce-4b79-b660-14185b13fc01.PNG)
![ilvl rework table](https://user-images.githubusercontent.com/99501957/196979886-b04a4a04-a034-4e7d-b21b-355ad092f51d.PNG)



## Steps to test these changes

1.  load in core change files
2.  Clean and rebuild Server to apply core changes
3.  Summon trusts with various levels of iLVL gear on and use !getstats to observe stat adjustments.
